src/databricks/labs/lsql/dashboards.py

```diff
@@ -1018,3 +1019,36 @@ def get_url(self, dashboard_id: str) -> str:
         # The /published redirects to the draft if the dashboard is not published
         dashboard_url = f"{self._ws.config.host}/dashboardsv3/{dashboard_id}/published"
         return dashboard_url
+
+
+class ExportDashboard(DashboardMetadata):
+    def __init__(self, sql_backend, config):
+        self._sql_backend = sql_backend
+        self._config = config
+        super().__init__(display_name="")
+
+    def export_dashboard(
+        self,
+        queries_path: Path,
+        export_path: Path,
+        catalog: str = "hive_metastore",
+        database: None = None,
+    ) -> None:
```

Comment:
instead of adding the class, create `def export_to_zipped_csv(self, sql_backend: SqlBackend) -> Path` on `DashboardMetadata` class. catalog/database replacement is already handled in that class.

Reply:
I'll refactor to follow your feedback thanks.

Reply:
Added a helper function to zip the CSV to avoid increasing the scope of the function.